.github/workflows/playground_selenium.yaml

```diff
@@ -2,38 +2,77 @@ name: Playground Selenium Tests
 
 on:
   workflow_dispatch:
+    inputs:
+      branch:
+        description: "Select which Git branch to run against"
+        required: true
+        default: "development"
+      network:
+        description: "Select the target environment"
+        required: true
+        type: choice
+        default: "dev"
+        options:
+          - local
+          - dev
+          - qa
+          - test
+          - main
   schedule:
     - cron: "0 6 * * *"
 
 jobs:
   selenium-run:
     runs-on: ubuntu-latest
+    env:
+      NETWORK: ${{ github.event.inputs.network }}
+
     steps:
-      - name: Checkout
+      - name: Checkout repository
         uses: actions/checkout@v4
-      - name: Setting up Python
+        with:
+          ref: ${{ github.event.inputs.branch }}
+
+      - name: Set up Python 3.10
         uses: actions/setup-python@v5
         with:
           python-version: "3.10"
-      - name: Installing all necessary packages
-        run: pip install -r  packages/playground/tests/frontend_selenium/requirements.txt
-      - name: Node install
+
+      - name: Install Python dependencies
+        run: |
+          pip install -r packages/playground/tests/frontend_selenium/requirements.txt
+          pip install pytest-html
+
+      - name: Set up Node.js 18
         uses: actions/setup-node@v4
         with:
-          node-version: 18
-      - name: Yarn install
-        run: yarn install
-      - name: Lerna Build
-        run: yarn lerna run build
-      - name: Yarn Serve
+          node-version: "18"
+
+      - name: Install JS dependencies & build
+        if: ${{ github.event.inputs.network == 'local' }}
+        run: |
+          yarn install
+          yarn lerna run build
+
+      - name: Start playground server
         run: make run project=playground &
-      - name: Wait for localhost
+
+      - name: Wait for server to be ready
         run: sleep 60
```

Comment:
All these steps for running the local server, and only the install step has the condition `if: ${{ github.event.inputs.network == 'local' }}`

Reply:
yes these steps are needed only for launch the dashboard locally; instead of this in all other network we just use the dashboard URL.

Reply:
Agreed to add a flag to be used if want a local build against any specific network, and by default the nightly workflow will run against staging env.